multiqc/modules/fgbio/ErrorRateByReadPosition.py

```diff
@@ -32,14 +35,24 @@ def parse_reports(self):
         for line in fh:
             fields = line.rstrip('\r\n').split('\t')
             fields[1:4] = [int(field) for field in fields[1:4]]
-            fields[4:] = [float(field) for field in fields[4:]]
+            fields[4:11] = [float(field) for field in fields[4:11]]
+            # Check if this is the new style
+            if len(fields) > 11:
```

Comment:
I'd first check in the header (i.e. `"collapse" in header` or whatever) to get the expected # of columns.  Then each row, I'd get the # of columns to keep and make sure the # of columns is the same across all rows.

Reply:
But the number of columns to keep depends on whether or not collapsed is True or not which we can't know until we actually start parsing the data OR implement something more complicated to peek ahead. 

I've changed it so that I check the header to establish which format type this is, and then assert that all lines match the header length. I do like getting the format from the header instead of nebulously checking each rows length. 

LMK if the new code is closer to what you are thinking.